### ISSUE TITLE
supergroup entries count (union/intersection) not updated when changing subgroups

### ISSUE THREAD

**Issue by ghost** [JabRef/jabref#10245]
### JabRef version

5.9 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 10 22H2

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Open attached library
2. Note 1 paper is set to "read', 1 set to "skimmed"
3. Create a group like this

![afbeelding](https://github.com/JabRef/jabref/assets/143431619/00f8dd3e-310e-438b-b597-d950a9b819bd)

4. Click "OK"
5. In jabref, the parent group "Lees status" (sorry I speak Dutch) has 0 entries, while it should have 2 (1 + 1, from the 2 subgroups). If you assign another "read" or "skimmed" status to a paper, the subgroup count is changed but NOT the count of the parent group "Lees status" (it always stays 0)

![afbeelding](https://github.com/JabRef/jabref/assets/143431619/981e1349-6a56-40e8-9997-220d6dcdf0b4)

[Vogels zijn dino's.zip](https://github.com/JabRef/jabref/files/12478133/Vogels.zijn.dino.s.zip)

### Appendix

...
<details>

<summary>Log File</summary>

```
Paste an excerpt of your log file here
```

</details>

**Comment by Siedlerchr** [comment]
Normally, this depends on the group hierarchy context (union or intersection)
https://docs.jabref.org/finding-sorting-and-cleaning-entries/groups#hierarchical-context

However, this seems to be an edge case with automatic keyword subgroups. With manual subgroups this works

**Comment by the-dino-girl** [comment]
> Normally, this depends on the group hierarchy context (union or intersection) https://docs.jabref.org/finding-sorting-and-cleaning-entries/groups#hierarchical-context
> 
> However, this seems to be an edge case with automatic keyword subgroups. With manual subgroups this works

Confirmed. It works with manual subgroups.

**Comment by the-dino-girl** [comment]
A small question about "intersection": if I have a supergroup which has 0 entries and a subgroup which has 5 entries, the result of intersection is always 0 entries? So there must be found entries that are both in the supergroup and the subgroup(s), meaning that the supergroup must have at least 1 entry to be able to have a possible intersection?

**Comment by ThiloteE** [comment]
### Steps to reproduce the behaviour

   1. Create a new group, below called the supergroup, and set it to union or intersection
   2. Drag-and-drop 1 or more subgroups to the supergroup
   3. The entries count of the supergroup is NOT updated at all. If you close and reload the library file, the entries count is correct.

**Comment by mkumar09** [comment]
> Normally, this depends on the group hierarchy context (union or intersection) https://docs.jabref.org/finding-sorting-and-cleaning-entries/groups#hierarchical-context
> 
> However, this seems to be an edge case with automatic keyword subgroups. With manual subgroups this works

Hi @Siedlerchr , I am already investigating this issue, Could you tell me what's a manual subgroup ? and any other useful information that might help in fixing this defect ?

**Comment by ThiloteE** [comment]
"Manual subgroups" in this context would refer to groups based on "Explicit selection". I do not know, if this issue can still be reproduced in the newest development version, so that would be the first thing to do. Test, if this is still a problem. @mkumar09 are you up for it?

You can assign yourself to this issue, by typing \/assign-me

**Comment by mkumar09** [comment]
> "Manual subgroups" in this context would refer to groups based on "Explicit selection". I do not know, if this issue can still be reproduced in the newest development version, so that would be the first thing to do. Test, if this is still a problem. [@mkumar09](https://github.com/mkumar09) are you up for it?
> 
> You can assign yourself to this issue, by typing /assign-me

Thanks!, I am able to reproduce this on latest main on Windows 11, Also what are automatic keyword subgroups? Is it all the other group selection apart from Explicit Selection?

**Comment by ThiloteE** [comment]
Yes, here, automatic keyword groups refers to all the other groups that automatically detect a keyword in an entry. More specifically all types of groups, where users will not have to add entries to groups in a manual way, but JabRef will do that automatically. A sub-group is a group that is hierarchically below another group, hence a sub-set of a set. For https://en.wikipedia.org/wiki/Set_(mathematics) for a explanation what is a set. At the moment, In JabRef, by default, entries in sub-groups are not automatically part of the top-level group, therefore the entry count is not required, but if you enable union/intersection, the entry count needs to be updated.

**Comment by Siva-Sai22** [comment]
/unassign-me 
I am having uni exams. Investigated and worked on this issue but the solution is not fully ready. 
Will continue after 3-4 days if nobody else takes it

**Comment by Yubo-Cao** [comment]
https://drive.google.com/file/d/1KMtBCBZLZSLTAyjBNqZqliJpbBMLy8ec/view?usp=sharing If I am not hallucinating I cannot reproduce this bug (but I am using CachyOS 7.3). Maybe the union table display itself is a bug?
